### DISCUSSION TITLE
Some GUI advice on detail

### DISCUSSION THREAD

**Discussion by THZrry** [SpartanJ/ecode#190] [Ideas]
Hello. There're some advice I think useful.
# 1 Button size
I think buttons'll be more comfortable to see if there're more pixels on height. Is it changeable with css? I may try later.
# 2 Menu ease in
Many widgets have their animations. Will menus have one? I think FireFox's multi-tab selector is an example on displaying.
# 3 Menu too
In directory tree, if I right click when there's one menu of it open, it looks like ths existing menu "move" to my mouse. I'm not sure it's ok or not ok, only mention it there. Moreover, I think if the main setting menu appears under the multi-tab bar will look more tidy? Hope I haven't expressed it strangely.
# 4 Editor cursor flashing
I think it's useful. But I translated this key, naming "cursor_blinking_time", but I can't find it. Is it hided?
# 5 Editor Minimap merging lines
I think the gap of minimap's lines is too long, making it seem a little strange. See that:
![comparing with sublime and vscode](https://github.com/SpartanJ/ecode/assets/140677303/e5072adb-f8eb-426a-93ef-4df7f55e6c3e)
# 6 Editor minimap triple clicking
Triple click with make one line selected. But it's only when release the third click. If I do not release the third click, it should keep selected on words, and when I drag, words may be selected by word-matching, not letter-matching. You can try this simpling click my words in browser.
# 7 Editor scroll by pixel
If do that, scrolling will be more smooth. Considering this GUI Frame is based on SDL2, this may be easy to make?

**Comment by SpartanJ** [comment]
Hi, thanks for the feedback, I'll give you my perspective:

1. I simply don't agree, they are high enough. Yes, it can be customised really easily overriding the CSS.

2. They are disabled, there's a very defined philosophy on how I accept animations: if the animation can delay the user interaction, then there will be no animation, if it's just a "decoration" then it is allowed, for menus, introducing animation is just introducing delay. You can add them if you like them by setting a transition in the CSS.

3. 
> In directory tree, if I right click when there's one menu of it open, it looks like ths existing menu "move" to my mouse.

IDK what you mean, menu appears just were you click.

> Moreover, I think if the main setting menu appears under the multi-tab bar will look more tidy? Hope I haven't expressed it strangely.

I've tried it and I don't like it. It looks like part of the current editor, which is confusing.

4. `Setting -> Document -> Global Settings -> Cursor Blinking Time` or just type `blink` on the command palette. 

5. I'm not sure what you mean, I really like how ecode's minimap looks, the other two examples look blurry an ugly to me. But I can say that I could add some config since the minimap already has a config but it's not exposes to the editor.

6.
> Triple click with make one line selected. But it's only when release the third click.
This one I agree. I don't use the mouse to code so I never noticed.

> If I do not release the third click, it should keep selected on words, and when I drag, words may be selected by word-matching, not letter-matching. You can try this simpling click my words in browser.

I don't agree with this one, for editing this should never be the behaviour. Browser behave like that because you're selecting read-only content. Usually if you triple-click select and drag it will keep selecting by line, not word or letter (check other editors).

7. Scrolling is already done by pixel. You can easily test it by scrolling from the scrollbar. What you probably want is a smooth animation when scrolling with the mouse wheel or the mimimap. This can be done and I'll probably add it optionally in the future, but I don't personally like it if forces an interpolation between two points (adds delay).

I appreciate the feedback. Thanks

**Comment by THZrry** [comment]
Your philosophy is meaningful, too. I wil try later.
4 I have checked my config and found it a big number.
6 I use triple clicking to select a paragraph of code so that I don't need to check if I selected half of the start or end word.
7 but touchboard behaves not like that. Could touchboard scroll by pixel?

**Comment by SpartanJ** [comment]
> 4 I have checked my config and found it a big number.

Oh, yes, I saw that bug, I think it wont happen to new users, I made some conversion mistake somewhere at some unknown version (at that it's being saved to the config file). You can fix it by just setting a new value. Time can be written in ms or s, for example: 600ms, 2s, etc.

> 6 I use triple clicking to select a paragraph of code so that I don't need to check if I selected half of the start or end word.

Sure, I understand!

> 7 but touchboard behaves not like that. Could touchboard scroll by pixel?

Yes, that's true, it's possible and it will be implemented at least for macOS, I'm not sure if Windows and Linux support scroll-wheel smoothing.

**Reply by SpartanJ** [reply]
BTW I improved triple clicking and it will be available in the next release, now the triple click will be triggered at the third mouse *down*, so it will let you continue selecting.

**Comment by THZrry** [comment]
There's another one that in window TextDnD support?
This may be easy, I made one in 2020 in python tkinter.

Ecode has not supported Draging in or out texts for now. I don't know if this is supported, TextDnD in window will be support at the same time. So will Ecode supporting TextDnD? It sounds like browsers' method any way.

**Reply by SpartanJ** [reply]
Text drag & drop is pending, yet another feature I will never use but I know some users use it. I'll try to implement it soon, it shouldn't be too tricky.
